src/main/java/net/snowflake/client/core/SFJsonResultSet.java

```diff
@@ -409,13 +407,24 @@ public Time getTime(int columnIndex) throws SFException {
       if (sfTime == null) {
         return null;
       }
+      if (resultSetSerializable.getUseSessionTimezone()) {
+        return new SnowflakeTimeWithSessionTimezone(
+            sfTime.getFractionalSeconds(ResultUtil.DEFAULT_SCALE_OF_SFTIME_FRACTION_SECONDS),
+            sfTime.getNanosecondsWithinSecond(),
+            resultSetSerializable.getUseSessionTimezone());
+      }
       return new Time(
```

Comment:
If we derive SFTime, SFDate, STTimestamp from sql.{Time,Date,Timestamp}, we might not need to return these original types anymore.

Reply:
This is true, I can take away the original Time type for this particular case. For Timestamp objects it can get a bit more tricky because we have 3 different Timestamp types with different offsets, and the original Timestamp needs to be used for 1 type while the new SnowflakeTimestamp needs to be used for the other 2.